examples/nucleo_f429zi/imu_adis16470/main.cpp

```diff
@@ -0,0 +1,128 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/platform/spi/spi_master_1.hpp>
+#include <modm/driver/inertial/adis16470.hpp>
+
+using namespace Board;
+
+using SpiMaster = SpiMaster1;
+using Mosi = GpioB5;
+using Miso = GpioB4;
+using Sck = GpioB3;
+using Cs = GpioA4;
+using Dr = GpioB9;
+using Rst = GpioB8;
+
+using Adis16470 = modm::Adis16470<SpiMaster, Cs>;
+
+Adis16470 imu{};
+
+int
+main()
+{
+	Board::initialize();
+	Leds::setOutput();
+
+	MODM_LOG_INFO << "ADIS16470 IMU demo on Nucleo-F429ZI\n" << modm::endl;
+
+	SpiMaster::initialize<Board::SystemClock, 650_kHz>();
+	SpiMaster::connect<Sck::Sck, Mosi::Mosi, Miso::Miso>();
+
+	Rst::setOutput();
+	Dr::setInput(Gpio::InputType::PullUp);
+
+	// Reset ADS816x chip
+	Rst::reset();
+	modm::delay(10ms);
+	Rst::set();
+
+	// Wait start-up time
+	modm::delay(252ms);
+
+	MODM_LOG_INFO << "Initializing IMU..." << modm::endl;
+	RF_CALL_BLOCKING(imu.initialize());
+
+	// Software reset
+	RF_CALL_BLOCKING(imu.writeGlobCmd(modm::adis16470::GlobCmd::SoftwareReset));
+	// Wait start-up time after reset
+	modm::delay(252ms);
+
+	// Trigger self test
+	RF_CALL_BLOCKING(imu.writeGlobCmd(modm::adis16470::GlobCmd::SensorSelfTest));
+	// Wait for self test to complete
+	modm::delay(14ms);
+
+	modm::adis16470::DiagStat_t diagStat = RF_CALL_BLOCKING(imu.readDiagStat());
+	MODM_LOG_INFO << "Reading DIAG_STAT register = " << diagStat << modm::endl;
+
+	modm::delay(1ms);
+	auto result = RF_CALL_BLOCKING(imu.readRegister(modm::adis16470::Register::PROD_ID));
+	if (result.has_value()) {
+		MODM_LOG_INFO << "ADIS16470 product id = " << *result << " (should be 16470==0x4056)" << modm::endl;
+	}
+	else {
+		MODM_LOG_INFO << "Unable to reag PROD_ID register" << modm::endl;
+	}
+
+	modm::adis16470::MscCtrl_t mscCtrl = RF_CALL_BLOCKING(imu.readMscCtrl());
+	MODM_LOG_INFO << "Reading MSC_CTRL register = " << mscCtrl << modm::endl;
+
+	// Output data rate 10Hz: 2000SPS/(199+1) = 10SPS
+	RF_CALL_BLOCKING(imu.writeRegister(modm::adis16470::Register::DEC_RATE, 199));
+
+	RF_CALL_BLOCKING(imu.writeRegister(modm::adis16470::Register::FILT_CTRL, 8)); // N_B = 2^8 = 64
+
+	mscCtrl = modm::adis16470::MscCtrl::DrPolarity;
+	// mscCtrl = modm::adis16470::MscCtrl::DrPolarity |
+	// 		  modm::adis16470::MscCtrl::LinearGCompensationGyro |
+	// 		  modm::adis16470::MscCtrl::PointOfPercussionAlign;
+	MODM_LOG_INFO << "Writing MSC_CTRL register = " << mscCtrl << modm::endl;
+	RF_CALL_BLOCKING(imu.writeMscCtrl(mscCtrl));
+
+	diagStat = RF_CALL_BLOCKING(imu.readDiagStat());
+	MODM_LOG_INFO << "Reading DIAG_STAT register = " << diagStat << modm::endl;
+
+	mscCtrl = RF_CALL_BLOCKING(imu.readMscCtrl());
+	MODM_LOG_INFO << "Reading MSC_CTRL register = " << mscCtrl << modm::endl;
```

Comment:
Somehow the write to MSC_CTRL is working, but also no error shows up in DIAG_STAT.

![image](https://user-images.githubusercontent.com/2820734/149400962-12abb457-760b-42cc-8fa6-71fb77df8459.png)

Reply:
The write looks fine. Have you tried to write and read back any other register the same way?

Could it be a timing issue? The wait times for the reset are listed as "typical" values.

Reply:
> Have you tried to write and read back any other register the same way?

Not yet.

> Could it be a timing issue? The wait times for the reset are listed as "typical" values.

Could be a timing issue, but the ADIS16470 seems to be ready and producing data even ~50ms earlier.
I'll add more delay.

Reply:
> The write looks fine. Have you tried to write and read back any other register the same way?

Other registers (e.g. `USER_SCR1`) read back fine.

> Could it be a timing issue? The wait times for the reset are listed as "typical" values.

Yes, waiting 3ms solves the issue and `MSC_CTRL` reads back correctly.
I added a 3ms timeout after writing `MSC_CTRL` in the `writeMscCtrl(MscCtrl_t v)` function https://github.com/modm-io/modm/blob/c0adf493e7f432b962dc9388bb3b5eeac9aa8d63/src/modm/driver/inertial/adis16470_impl.hpp#L144-L147